src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -507,18 +515,17 @@ FinalState finalState
                             => finalState,
 
                         // We've got nothing.  Build it from scratch :(
-                        NoCompilationState
+                        NoCompilationState noCompilationState
                             => await BuildFinalStateFromScratchAsync(
                                 compilationState,
-                                state.GeneratorInfo,
+                                noCompilationState,
```

Comment:
all these functions became much clearer by stating explicitly what input-tracker-state they operate on, and waht explicit output-tracker-state they return.  That also prevents accidentally using the wrong function in the wrong location.